src/librustc_back/target/windows_base.rs

```diff
@@ -63,7 +60,32 @@ pub fn opts() -> TargetOptions {
 
             // Always enable DEP (NX bit) when it is available
             "-Wl,--nxcompat".to_string(),
+
+            // Do not use the standard system startup files or libraries when linking
+            "-nostdlib".to_string(),
+        ),
+        pre_link_objects_exe: vec!(
+            "crt2.o".to_string(),
+            "rsbegin.o".to_string(),
+        ),
+        pre_link_objects_dll: vec!(
+            "dllcrt2.o".to_string(),
+            "rsbegin.o".to_string(),
+        ),
+        late_link_args: vec!(
+            "-lmingwex".to_string(),
+            "-lmingw32".to_string(),
+            "-lgcc".to_string(), // alas, mingw* libraries above depend on libgcc
+            "-lmsvcrt".to_string(),
+            "-ladvapi32".to_string(),
+            "-lshell32".to_string(),
+            "-luser32".to_string(),
+            "-lkernel32".to_string(),
```

Comment:
Yeah without `-nodefaultlibs` the linker passes in things like `-lSystem` on OSX and `-lc` on other unixes, but we in general like to have a handle on our dependencies in Rust. For example it's the crate's job to depend on a native library rather than the compiler injecting it for you. Along those lines I think it'd be fine to remove msvcrt, shell32, kernel32, advapi32, and user32 at the very least. If link errors arise then those annotations can be added to the standard library/liblibc as apporpriate.

I think that `-lgcc` can be omitted entirely as it looks like those are just GCC intrinsics (e.g. this shouldn't cause link errors). The `mingw32` library looks like it may also be able to get omitted? Perhaps worth at least trying to omit it! The `mingwex` library looks a little more serious and may need to be injected by the compiler anyway. Regardless though it in theory shouldn't be required here (but rather as part of libstd/rtdeps.rs) if it's actually required.